### PR TITLE
TST: including peer forwarder in e2e servicemap test

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*

*Description of changes:*
- Refactors some setup functions in endToEndServiceMapTest
- Add an additional instance of data-prepper in gradle docker
- Included peerforwarder in the data-prepper pipeline.yml

Note: not sure how to handle create arbitrary number of data prepper containers in gradle task w/o hardcoding. Since it is not blocking, will leave it to future if necessary.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-prepper-core/src/integrationTest/resources/pipeline.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 entry-pipeline:
   source:
     otel_trace_source:
+  processor:
+    - peer_forwarder:
+        discovery_mode: "static"
+        data_prepper_ips: ["data-prepper1", "data-prepper2"]
```
Comment (wrijeff) [inline data-prepper-core/src/integrationTest/resources/pipeline.yml:7]: I'm making a minor change by renaming this property to "static_endpoints" in #222. The intent is to make it more clear that either hostnames _or_ IP addresses are valid for the static discovery mode. Sorry for not letting you know sooner.

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java [hunk 2]
```diff
@@ -34,28 +37,38 @@
...
         );
 
-        sendExportTraceServiceRequestToSource(exportTraceServiceRequest1);
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_1, exportTraceServiceRequest11);
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_2, exportTraceServiceRequest12);
 
         //Verify data in elasticsearch sink
+        final List<ServiceMapTestData> testDataSet1 = Stream.of(testDataSet11, testDataSet12)
+                .flatMap(Collection::stream).collect(Collectors.toList());
+        final List<Map<String, Object>> possibleEdges = getPossibleEdges(TEST_TRACEID_1, testDataSet1);
```
Comment (AustinTag) [inline data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java:71]: I think this is the proper way of implicitly verifying the peer forwarding (for service map), good job.

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java [hunk 3]
```diff
@@ -172,44 +194,51 @@ public static ExportTraceServiceRequest getExportTraceServiceRequest(List<Resour
...
+                    destination.put("domain", currData.serviceName);
                     Map<String, Object> edge = new HashMap<>();
-                    edge.put("serviceName", parentServiceName);
-                    edge.put("kind", parentRS.getInstrumentationLibrarySpans(0).getSpans(0).getKind().name());
+                    edge.put("serviceName", parentData.serviceName);
+                    edge.put("kind", parentData.spanKind.name());
                     edge.put("traceGroupName", rootSpanName);
                     edge.put("destination", destination);
                     edge.put("target", null);
                     possibleEdges.add(edge);
```
Comment (AustinTag) [inline data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java:220]: Nit: Nothing incorrect here, but I think this code could probably use some cleanup in terms of separating large chunks of code into logical functions. For example:

getEdgeMap(curData, parentData)

File: data-prepper-core/integrationTest.gradle [hunk 4]
```diff
@@ -68,24 +68,54 @@ task buildDataPrepperDockerImage(type: DockerBuildImage) {
...
+    doLast {
+        sleep(10*1000)
+    }
+}
+
+task stopDataPrepper1DockerContainer(type: DockerStopContainer) {
+    targetContainerId createDataPrepper1DockerContainer.getContainerId()
+}
+
+task stopDataPrepper2DockerContainer(type: DockerStopContainer) {
```
Comment (AustinTag) [inline data-prepper-core/integrationTest.gradle:109]: Nit: This works for now, but we may want to look into a way of calling a task with an argument if we can? For example:

startDataPrepperContainers(int[] ports)

That way we can scale up testing and re-use these functions

**Review by AustinTag — APPROVED** [review]
LGTM, just a few nitpicks (not required, just thinking for the future)

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java (hunk 2 above)
Comment (chenqi0805) [reply data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java:71]: Credits to @kowshikn

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java (hunk 3 above)
Comment (chenqi0805) [reply data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java:220]: Will take another look.

File: data-prepper-core/integrationTest.gradle (hunk 4 above)
Comment (chenqi0805) [reply data-prepper-core/integrationTest.gradle:109]: Agreed. We probably need to define new classes in gradle for this purpose. For the purpose of this test it is not worth the efforts. Will re-examine when it becomes necessary.

File: data-prepper-core/src/integrationTest/resources/pipeline.yml (hunk 1 above)
Comment (kowshikn) [reply data-prepper-core/src/integrationTest/resources/pipeline.yml:7]: Now one of you should deal with merge conflict :P

**Review by kowshikn — APPROVED** [review]
LGTM. Work with jeff on the conflict alone.

**Comment by chenqi0805** [comment]
> LGTM. Work with jeff on the conflict alone.

Sure. Definitely within the scope of this PR.

**Pull request merged**